components/nexusoperations/workflow/commands.go

```diff
@@ -110,6 +111,14 @@ func (ch *commandHandler) HandleScheduleCommand(
 		}
 	}
 
+	if err := timestamp.ValidateProtoDuration(attrs.ScheduleToCloseTimeout); err != nil {
```

Comment:
How does this behave if the timeout is 0? Shouldn't we treat 0 as unset?

Reply:
In the state machine we have this check:

```go
	if o.ScheduleToCloseTimeout.AsDuration() != 0 {
		return []hsm.Task{TimeoutTask{deadline: o.ScheduledTime.AsTime().Add(o.ScheduleToCloseTimeout.AsDuration())}}, nil
	}
```

Reply:
0 is valid in this check.

FWIW calling `.AsDuration()` does a best effort conversion, so there are some small edge cases where it could return a negative duration.